client/src/components/Reservation.jsx

```diff
@@ -32,76 +33,74 @@ class Reservation extends React.Component {
   }
 
   componentDidMount() {
-    this.fetch();
+    this.getAvailabilityInfo(this.state.date, this.state.time, this.state.party);
   }
 
   setName(name) {
     this.setState({ name });
   }
 
-  requestReservation(time) {
-    // console.log('requestReservation', this.props.id, time, this.state.party);
-    helper.requestReservation(
-      this.props.id, this.state.date, time, this.state.name, this.state.party,
-      (err) => {
-        if (!err) {
-          // success
-          console.log('requestReservation success');
-        } else {
-          // error
-          console.log('requestReservation error');
-        }
-      },
-    );
-  }
-
   getAvailabilityInfo(date, time, party) {
-    console.log('getAvailabilityInfo', this.props.id, date, time, party);
-    console.log('state vars', this.state.date, this.state.time, this.state.party);
     helper.getReservationInfo(this.props.id, date, (err, data) => {
       if (!err) {
         console.log('getAvailabilityInfo', data);
         this.setState({
+          message: '',
           availabilityInfo: data.reservations,
           bookingsMadeToday: data.madeToday,
           party,
           date,
           time,
         });
       } else {
-        console.log('getAvailabilityInfo', err);
+        this.setState({
+          message: `Error getting availability info: ${err}`,
+        });
       }
     });
   }
 
-  fetch(date = (new Date()).toISOString().slice(0, 10)) {
-    helper.getReservationInfo(this.props.id, date, (err, data) => {
-      console.log('getReservationInfo callback', this.props.id, data);
-      console.log(this.state.bookingsMadeToday, this.state.availabilityInfo);
-      this.setState({
-        bookingsMadeToday: data.madeToday,
-        availabilityInfo: data.reservations,
-      });
-    });
+  requestReservation(time) {
+    helper.requestReservation(
+      this.props.id, this.state.date, time, this.state.name, this.state.party,
+      (err) => {
+        if (!err) {
+          console.log('requestReservation success');
+          this.setState({
+            message: 'Your table has been saved!',
+          });
+        } else {
+          this.setState({
+            message: `Error making reservation: ${err}`,
+          });
+        }
+      },
+    );
   }
 
-
   render() {
+    const {
+      name, bookingsMadeToday, availabilityInfo, message,
+    } = this.state;
+
```

Comment:
Cool. So this allows you to simply refer to 'name' when passing it to SetName instead of using 'this.state.name'?

Reply:
yup!  Just a shorthand that makes it a little easier to read